frontend/src/pages/documentation/static-dataflow-analysis.mdx

```diff
@@ -0,0 +1,156 @@
+import JavaKotlinCodeBlock from "../../components/JavaKotlinCodeBlock";
+import Examples from "../../components/examples";
+
+# Static dataflow analysis
+
+With the `jacodb-analysis` module, one can perform static dataflow analysis based on the three-address code intermediate representation.
+It implements the IFDS solver, described in the <a href="https://dx.doi.org/10.1145/199448.199462">Reps, Horwitz, and Sagiv (1995)</a> paper.
+IFDS is the acronym for the class of the _interprocedural_, _finite_, _distributive_, _subset_ problems.
+
+The `jacodb-analysis` module provides you with several ready-to-use analyses along with an API to build your own analyses.
+
+Current implementation splits code into `Units`, so that the IFDS framework can analyse them concurrently.
+Information is shared between the units via `Summaries`, but the lifecycle of each unit is controlled
+separately. This makes the implementation highly scalable and still precise.
+
+## Basic usage
+
+You can perform analysis right from your code or via command-line interface.
+
+### Calling from your code
+
+The analysis entry point is the `runAnalysis` method called from `AnalysisMain`.
+It takes the following parameters:
+* `graph` — an application graph that is used for analysis, the _supergraph_ in terms of the <a href="https://dx.doi.
+org/10.1145/199448.199462">original paper</a>.
+To obtain this graph, one should call the `newApplicationGraphForAnalysis` method from `ApplicationGraphFactory`.
+* `unitResolver` — an object that groups methods into units. Find more about <a href="#unit-resolvers">unit resolvers</a>.
+* `ifdsUnitRunner` — a runner instance, which is used to analyze each unit. This is what defines concrete analysis.
+Ready-to-use runners are located in `RunnersLibrary`.
+* `methods` — a list of methods to analyze.
+* `timeoutMillis` — an optional timeout (in milliseconds).
+
+For example, to detect the unused variables in the given `analyzedClass` methods, you may run the following code
+(assuming that `classpath` is an instance of `JcClasspath`):
+
+<JavaKotlinCodeBlock
+    javaCode={Examples.runAnalysisExample.java}
+    kotlinCode={Examples.runAnalysisExample.kotlin}
+/>
+
+### Using command-line interface
+
+To perform analyses, one can use command-line interface (CLI) from the `jacodb-cli` module.
+Specify the following arguments:
+* `—analysisConf, -a` — a path to a file with the analyses configuration in the JSON format (find detailed description below).
+* `—start, -s` — classes to start the analyses from.
+* `—classpath, -cp` — a classpath necessary for JacoDB analyses.
+* `[optional] —dbLocation, -l` — location of SQLite database to store bytecode data.
+If not specified, no data will be stored in the database.
+* `[optional] —output, -o` — a file to store the analysis report. It defaults to `report.json`.
```

Comment:
Please, revert long dash to double hyphen in fullname arguments specification, i.e. use `--analysisConf, -a` instead of `—analysisConf, -a`, and same for other arguments